tests/core/identity/identify_push/test_identify_push.py

```diff
@@ -0,0 +1,431 @@
+    3. Both host_b and host_c have their peerstores updated with host_a's information
+
+    This tests the broadcasting capability of the identify/push protocol.
+    """
+    # Create three hosts
+    async with host_pair_factory(security_protocol=security_protocol) as (
+        host_a,
+        host_b,
+    ):
+        # Create a third host
+        # Instead of using key_pair, create a new host directly
+        import multiaddr
+
+        from libp2p import (
+            new_host,
+        )
+        from libp2p.crypto.secp256k1 import (
+            create_new_key_pair,
+        )
+        from libp2p.peer.peerinfo import (
+            info_from_p2p_addr,
+        )
+
+        # Create a new key pair for host_c
+        key_pair_c = create_new_key_pair()
+        host_c = new_host(key_pair=key_pair_c)
+
+        # Set up the identify/push handlers
+        host_b.set_stream_handler(ID_PUSH, identify_push_handler_for(host_b))
+        host_c.set_stream_handler(ID_PUSH, identify_push_handler_for(host_c))
+
+        # Start listening on a random port using the run context manager
+        listen_addr = multiaddr.Multiaddr("/ip4/127.0.0.1/tcp/0")
+        async with host_c.run([listen_addr]):
+            # Connect host_c to host_a and host_b
+            await host_c.connect(info_from_p2p_addr(host_a.get_addrs()[0]))
+            await host_c.connect(info_from_p2p_addr(host_b.get_addrs()[0]))
+
+            # Push identify information from host_a to all connected peers
+            from libp2p.identity.identify_push.identify_push import (
+                push_identify_to_peers,
+            )
+
+            await push_identify_to_peers(host_a)
+
+            # Wait a bit for the push to complete
+            await trio.sleep(0.1)
+
+            # Check that host_b's peerstore has been updated with host_a's information
+            peerstore_b = host_b.get_peerstore()
+            peer_id_a = host_a.get_id()
+
+            # Check that the peer is in the peerstore
+            assert peer_id_a in peerstore_b.peer_ids()
+
+            # Check that host_c's peerstore has been updated with host_a's information
+            peerstore_c = host_c.get_peerstore()
+
+            # Check that the peer is in the peerstore
+            assert peer_id_a in peerstore_c.peer_ids()
+
+
+@pytest.mark.trio
+async def test_push_identify_to_peers_with_explicit_params(security_protocol):
+    """
+    Test the push_identify_to_peers function with explicit parameters.
+
+    This test verifies that:
+    1. The function correctly handles an explicitly provided set of peer IDs
+    2. The function correctly uses the provided observed_multiaddr
+    3. The identify information is only pushed to the specified peers
+    4. The observed address is correctly included in the identify message
+
+    This test ensures all parameters of push_identify_to_peers are properly tested.
+    """
+    import multiaddr
```

Comment:
Any reason not to import these just once at the top? Same imports are done within `test_identify_push_to_peers`